lib/kernel/src/file.erl

```diff
@@ -398,24 +398,32 @@ write_file(Name, Bin) ->
       Reason :: posix() | badarg | terminated | system_limit.
 
 write_file(Name, Bin, ModeList) when is_list(ModeList) ->
-    case make_binary(Bin) of
-	B when is_binary(B) ->
-	    case open(Name, [binary, write | 
-			     lists:delete(binary, 
-					  lists:delete(write, ModeList))]) of
-		{ok, Handle} ->
-		    case write(Handle, B) of
-			ok ->
-			    close(Handle);
-			E1 ->
-			    _ = close(Handle),
-			    E1
-		    end;
-		E2 ->
-		    E2
-	    end;
-	E3 ->
-	    E3
+  case lists:member(raw, ModeList) of
+    true ->
+        do_write_file(Name, Bin, ModeList);
+    false ->
+        case make_binary(Bin) of
+            B when is_binary(B) ->
+                do_write_file(Name, B, ModeList);
+            Error ->
+                Error
+        end
+  end.
+
+do_write_file(Name, B, ModeList) ->
```

Comment:
@lexmag I renamed some variables - see what you think. (I went with `IOData` instead of `Data`.)

@fenollp Good idea about adding a spec. Can you tell me where to do that?